server/service/integration_core_test.go

```diff
@@ -2756,6 +2815,92 @@ func (s *integrationTestSuite) TestPaginateListSoftware() {
 	assertResp(lsResp, nil, time.Time{})
 }
 
+func (s *integrationTestSuite) TestChangeUserEmail() {
+	t := s.T()
+
+	// create a new test user
+	user := &fleet.User{
+		Name:       t.Name(),
+		Email:      "testchangeemail@example.com",
+		GlobalRole: ptr.String(fleet.RoleObserver),
+	}
+	userRawPwd := "foobarbaz1234!"
+	err := user.SetPassword(userRawPwd, 10, 10)
+	require.Nil(t, err)
+	user, err = s.ds.NewUser(context.Background(), user)
+	require.Nil(t, err)
+
+	// try to change email with an invalid token
+	var changeResp changeEmailResponse
+	s.DoJSON("GET", "/api/v1/fleet/email/change/invalidtoken", nil, http.StatusNotFound, &changeResp)
+
+	// create a valid token for the test user
+	err = s.ds.PendingEmailChange(context.Background(), user.ID, "testchangeemail2@example.com", "validtoken")
+	require.Nil(t, err)
+
+	// try to change email with a valid token, but request made from different user
+	changeResp = changeEmailResponse{}
+	s.DoJSON("GET", "/api/v1/fleet/email/change/validtoken", nil, http.StatusNotFound, &changeResp)
+
+	// switch to the test user and make the change email request
+	s.token = s.getTestToken(user.Email, userRawPwd)
+	defer func() { s.token = s.getTestAdminToken() }()
+
+	changeResp = changeEmailResponse{}
+	s.DoJSON("GET", "/api/v1/fleet/email/change/validtoken", nil, http.StatusOK, &changeResp)
+	require.Equal(t, "testchangeemail2@example.com", changeResp.NewEmail)
+
+	// using the token consumes it, so making another request with the same token fails
+	changeResp = changeEmailResponse{}
+	s.DoJSON("GET", "/api/v1/fleet/email/change/validtoken", nil, http.StatusNotFound, &changeResp)
+}
+
+func (s *integrationTestSuite) TestSearchTargets() {
+	t := s.T()
+
+	hosts := s.createHosts(t)
+
+	lblIDs, err := s.ds.LabelIDsByName(context.Background(), []string{"All Hosts"})
+	require.NoError(t, err)
+	require.Len(t, lblIDs, 1)
+
+	// no search criteria
+	var searchResp searchTargetsResponse
+	s.DoJSON("POST", "/api/v1/fleet/targets", searchTargetsRequest{}, http.StatusOK, &searchResp)
+	require.Equal(t, uint(0), searchResp.TargetsCount)
+	require.Len(t, searchResp.Targets.Hosts, len(hosts)) // the HostTargets.HostIDs are actually host IDs to *omit* from the search
```

Comment:
> the HostTargets.HostIDs are actually host IDs to *omit* from the search

So this got me *way* confused and spending more time than I would've wanted trying to understand the results. It is pretty clear once you look at the datastore layer that gets called with those IDs: https://github.com/fleetdm/fleet/blob/main/server/datastore/mysql/hosts.go#L772-L778

But then at the endpoint layer, the field is called `searchTargetsRequest.Selected`, making that really counter-intuitive. But based on how the call is made in the service, it seems deliberate to pass those IDs as the ones to **omit** from the queries: https://github.com/fleetdm/fleet/blob/main/server/service/service_targets.go#L33-L50 . The API docs don't really mention exactly what is and isn't part of the returned payload? https://github.com/fleetdm/fleet/blob/main/docs/01-Using-Fleet/03-REST-API.md#targets

Given that, it looks unlikely that this is a bug, just a behaviour that is not super obvious (or wasn't to me, anyway). If I can get confirmation that this is indeed working as intended, I will at least at a comment on the `searchTargetsRequest.Selected` field.

Reply:
> the field is called searchTargetsRequest.Selected, making that really counter-intuitive

Right, this is from the UI perspective, when you select a host, don't return it the next time it calls /targets, as you already have it... but agreed it's confusing!